main.go

```diff
@@ -42,88 +45,408 @@ func main() {
 	tc := oauth2.NewClient(ctx, ts)
 	client := github.NewClient(tc)
 
-	// Get the Pull Request details
+	// Fetch PR details
 	pr, _, err := client.PullRequests.Get(ctx, *owner, *repo, *prNumber)
 	if err != nil {
 		fmt.Printf("Error fetching PR details: %v\n", err)
 		os.Exit(1)
 	}
 
-	// Generate a review using GPT-4
-	review, err := generateReviewWithAssistant(pr)
+	// Fetch the current user (the reviewer)
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		fmt.Printf("Error fetching user details: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Fetch PR checks (e.g., CI tests)
+	checks, _, err := client.Checks.ListCheckRunsForRef(ctx, *owner, *repo, *pr.Head.SHA, &github.ListCheckRunsOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PR checks: %v\n", err)
+		os.Exit(1)
+	}
+
+	// If any check has failed, do not allow approval
+	checksPassed := true
+	for _, check := range checks.CheckRuns {
+		if check.GetConclusion() == "failure" {
+			checksPassed = false
+			break
+		}
+	}
+
+	// Fetch PR files
+	files, _, err := client.PullRequests.ListFiles(ctx, *owner, *repo, *prNumber, &github.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PR files: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Check for pending reviews
+	pendingReview, err := getPendingReview(client, ctx, *owner, *repo, *prNumber)
+	if err != nil {
+		fmt.Printf("Error checking for pending reviews: %v\n", err)
+		os.Exit(1)
+	}
+
```

Comment:
** Consider verifying that `files` is not empty before proceeding to create comments to avoid null pointer dereference.